public/js/ClusterPlot.js

```diff
@@ -0,0 +1,166 @@
+var d3 = require('d3');
+require('d3-tip')(d3);
+
+function ClusterPlot(sampleData) {
```

Comment:
Ok, super-confused here... `ClusterPlot` is a function, an object, both, a constructor? `this` refers to what?